mono/mini/mini-runtime.c

```diff
@@ -1429,10 +1429,11 @@ mono_resolve_patch_target (MonoMethod *method, MonoDomain *domain, guint8 *code,
 	case MONO_PATCH_INFO_GC_SAFE_POINT_FLAG:
 #if defined(__native_client_codegen__)
 		target = (gpointer)&__nacl_thread_suspension_needed;
-#elif defined (USE_COOP_GC)
-		target = (gpointer)&mono_polling_required;
 #else
-		g_error ("Unsuported patch target");
+		if (mono_threads_is_coop_enabled ())
+			target = (gpointer)&mono_polling_required;
+		else
+			g_error ("Unsuported patch target");
```

Comment:
Wouldn't it be simpler to always insert safepoints (in mono_insert_safepoints) ? As the check for coop is done at runtime anyway, and mono_polling_required will never be set to 1 in case of the coop suspend is not used.
That would add a branch which would never be used when there is no coop, but don't we want to have it the default anyway?